frontend/src/components/CustomTimePicker/timezoneUtils.ts

```diff
@@ -0,0 +1,143 @@
+import { getTimeZones } from '@vvo/tzdb';
+import dayjs from 'dayjs';
+import timezone from 'dayjs/plugin/timezone';
+import utc from 'dayjs/plugin/utc';
+
+dayjs.extend(utc);
+dayjs.extend(timezone);
+
+export interface Timezone {
+	name: string;
+	value: string;
+	offset: string;
+	searchIndex: string;
+	hasDivider?: boolean;
+}
+
+// Constants
+const TIMEZONE_TYPES = {
+	BROWSER: 'BROWSER',
+	UTC: 'UTC',
+	STANDARD: 'STANDARD',
+} as const;
+
+type TimezoneType = typeof TIMEZONE_TYPES[keyof typeof TIMEZONE_TYPES];
+
+const UTC_TIMEZONE: Timezone = {
+	name: 'Coordinated Universal Time — UTC, GMT',
+	value: 'UTC',
+	offset: 'UTC',
+	searchIndex: 'UTC',
+	hasDivider: true,
+};
+
+// Helper functions
+const isValidTimezone = (tzName: string): boolean => {
+	try {
+		dayjs.tz(dayjs(), tzName);
```

Comment:
add comments to the code.

Reply:
@YounixM, this is not needed, removing it. I figured out how to get timezones without using the library i.e. using `Intl.supportedValuesOf('timeZone')`.